segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/SegmentMetadataUpdateTransaction.java

```diff
@@ -455,6 +461,28 @@ void preProcessAsSourceSegment(MergeSegmentOperation operation) throws StreamSeg
         }
     }
 
+    /**
+     * Returns the total number of unique ExtendedAttributes that would exist for the segment assuming the
+     * {@link AttributeUpdateCollection} would be applied.
+     *
+     * @param attributeUpdates The {@link AttributeUpdateCollection} to apply.
+     * @return The number of unique extended attributes.
+     */
+    private int getUniqueExtendedAttributesCount(AttributeUpdateCollection attributeUpdates) {
+        int count = 0;
+        for (Map.Entry<AttributeId, Long> attribute : baseAttributeValues.entrySet()) {
```

Comment:
To speed things up, I suggest we just get the size of the `baseAttributeValues`. Even though that's going to contain core attributes as well, this code is on the hot path for every single append in the segment store so we must do everything we can to make it as fast as possible. We can always increase our attribute limit to 32 and use `.size()` here to approximate the number of attributes.